### PR TITLE
Move Ssrc -> (Mid, Ssrc) mappping to `Streams`

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

File: src/streams/mod.rs [hunk 1]
```diff
@@ -441,15 +484,109 @@ impl Streams {
...
+        // Add mapping for new main SSRC
+        self.associate_ssrc_mid(ssrc_to, mid, ssrc_to, None);
+
+        // Map the RTX to the new main SSRC
+        if let Some(rtx) = rtx {
+            self.associate_ssrc_mid(rtx, mid, ssrc_to, None);
+        }
     }
+
+    pub(crate) fn debug(&self) {
```
Comment (k0nserv) [inline src/streams/mod.rs:516]: Hmm this was supposed to be excluded...

File: src/streams/mod.rs [hunk 2]
```diff
@@ -441,15 +484,109 @@ impl Streams {
         }
     }
 
-    pub(crate) fn change_stream_rx_ssrc(&mut self, ssrc_from: Ssrc, ssrc_to: Ssrc) {
+    pub(crate) fn change_stream_rx_ssrc(
+        &mut self,
+        ssrc_from: Ssrc,
+        ssrc_to: Ssrc,
+        rtx_to: Option<Ssrc>,
+    ) {
```
Comment (algesten) [inline src/streams/mod.rs:492]: I prefer having a separate function for changing RTX, this overloads the purpose a bit too much.

**Review by algesten — APPROVED** [review]
Just one thing

File: src/streams/mod.rs [hunk 3]
```diff
@@ -198,16 +214,42 @@ impl Streams {
...
+
+            // SSRC is mapped to a Sender/Receiver in this media. Make an entry for it.
+            associate_ssrc_mid(
+                &mut self.source_keys_rx,
+                ssrc,
+                mid,
+                ssrc_main,
+                Some("Known SSRC"),
+            );
+            self.debug();
```
Comment (k0nserv) [inline src/streams/mod.rs:245]: Also should go

File: src/streams/mod.rs (hunk 2 above)
Comment (k0nserv) [reply src/streams/mod.rs:492]: Aight, we have a quite a lot of methods that take an optional `rtx: Option<Ssrc>` so I tried to mirror that

**Review by algesten — APPROVED** [review]

**Pull request merged**